build.gradle.kts

```
@@ -0,0 +1,214 @@
+    id("com.gradle.plugin-publish") version "1.2.0"
+    id("com.github.breadmoirai.github-release") version "2.4.1"
+    id("org.gradle.wrapper-upgrade") version "0.11.1"
+}
+
+val releaseVersion = releaseVersion()
+val releaseNotes = releaseNotes()
+val isCI = (System.getenv("CI") ?: "false").toBoolean()
+
+group = "org.gradle.android"
+version = releaseVersion.get()
+description = "A Gradle plugin to fix Android caching problems"
+
+repositories {
+    google()
+    mavenCentral()
+}
+
+dependencies {
+    val versions = mapOf(
+        "agp" to "7.3.1",
+        "sdkBuildTools" to "30.0.4",
+        "spock" to "2.3-groovy-3.0",
+    )
+
+    compileOnly(gradleApi())
+    compileOnly("com.android.tools.build:gradle:${versions["agp"]}")
+    compileOnly("com.android.tools:common:${versions["sdkBuildTools"]}")
+    compileOnly("com.android.tools:sdk-common:${versions["sdkBuildTools"]}")
+    implementation("com.google.guava:guava:31.1-jre")
+
+
+    testImplementation(gradleTestKit())
+    testImplementation("com.android.tools.build:gradle:${versions["agp"]}")
+    testImplementation(platform("org.spockframework:spock-bom:${versions["spock"]}"))
+    testImplementation("org.spockframework:spock-core") { exclude(group = "org.codehaus.groovy") }
+    testImplementation("org.spockframework:spock-junit4") { exclude(group = "org.codehaus.groovy") }
+    testImplementation("org.junit.jupiter:junit-jupiter-api")
+}
+
+wrapperUpgrade {
+    gradle {
+        create("android-cache-fix-gradle-plugin") {
+            repo.set("gradle/android-cache-fix-gradle-plugin")
+        }
+    }
+}
+
+java {
+    toolchain {
+        // AGP 7+ only supports JDK 11+
+        languageVersion.set(JavaLanguageVersion.of(11))
+    }
+}
+
+// Main plugin publishing metadata
+gradlePlugin {
+    website.set("https://github.com/gradle/android-cache-fix-gradle-plugin")
+    vcsUrl.set("https://github.com/gradle/android-cache-fix-gradle-plugin")
+
+    plugins {
+        create("androidCacheFixPlugin") {
+            id = "org.gradle.android.cache-fix"
+            displayName = "Gradle Android cache fix plugin"
+            description = releaseNotes.get()
+            implementationClass = "org.gradle.android.AndroidCacheFixPlugin"
+            tags.addAll("android", "cache", "fix")
+        }
+    }
+}
+
+// A local repo we publish our library to for testing in order to workaround limitations
+// in the TestKit plugin classpath.
+val localRepo = file("$buildDir/local-repo")
+
+val isProdPortal = System.getProperty("gradle.portal.url") != null
+// The legacy groupId gradle.plugin.* is only allowed when the plugin
+// has already been published
+val pluginGroupId = if (isCI && isProdPortal) "gradle.plugin.org.gradle.android" else project.group
+publishing {
+    publications {
+        create<MavenPublication>("pluginMaven") {
+            groupId = pluginGroupId.toString()
+        }
+    }
+    repositories {
+        maven {
+            url = uri(localRepo.toURI())
```

can this be simplified? it seems somehow redundant.
```suggestion
            url = localRepo.toURI()
```